src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskCompleteByIdControllerTest.java

```diff
@@ -194,9 +250,317 @@ public void should_return_a_403_when_the_user_did_not_have_sufficient_permission
+    }
+
+    @Test
+    public void should_return_a_204_if_task_was_not_previously_assigned_and_assign_and_complete_true() {
+        TestVariables taskVariables = common.setupTaskAndRetrieveIds();
+        String taskId = taskVariables.getTaskId();
+
+        common.setupOrganisationalRoleAssignment(authenticationHeaders);
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            taskId,
+            new CompleteTaskRequest(new CompletionOptions(true)),
+            authenticationHeaders
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.NO_CONTENT.value());
+
+        assertions.taskVariableWasUpdated(taskVariables.getProcessInstanceId(), "taskState", "completed");
+
+        common.cleanUpTask(taskId, REASON_COMPLETED);
+    }
+
+    @Test
+    public void should_return_a_204_when_completing_a_task_with_restricted_role_assignment_assign_and_complete_true() {
+        TestVariables taskVariables = common.setupTaskAndRetrieveIds();
+        String taskId = taskVariables.getTaskId();
+
+        common.setupRestrictedRoleAssignment(taskVariables.getCaseId(), authenticationHeaders);
+        given.iClaimATaskWithIdAndAuthorization(
+            taskId,
+            authenticationHeaders
+        );
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            taskId,
+            new CompleteTaskRequest(new CompletionOptions(true)),
+            authenticationHeaders
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.NO_CONTENT.value());
+
+        assertions.taskVariableWasUpdated(taskVariables.getProcessInstanceId(), "taskState", "completed");
+
+        common.cleanUpTask(taskId, REASON_COMPLETED);
+    }
+
+    @Test
+    public void should_return_a_401_when_the_user_did_not_have_any_roles_and_assign_and_complete_true() {
+        TestVariables taskVariables = common.setupTaskAndRetrieveIds();
+        String taskId = taskVariables.getTaskId();
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            taskId,
+            new CompleteTaskRequest(new CompletionOptions(true)),
+            authenticationHeaders
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.UNAUTHORIZED.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("timestamp", lessThanOrEqualTo(ZonedDateTime.now().plusSeconds(60)
+                .format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT))))
+            .body("error", equalTo(HttpStatus.UNAUTHORIZED.getReasonPhrase()))
+            .body("status", equalTo(HttpStatus.UNAUTHORIZED.value()))
+            .body("message", equalTo("User did not have sufficient permissions to perform this action"));
+
+        common.cleanUpTask(taskId, REASON_COMPLETED);
+    }
+
+    @Test
+    public void should_return_a_403_when_user_jurisdiction_did_not_match_and_assign_and_complete_tru() {
```

Comment:
very very minor. Just noticed the test name last word missing an 'e'

Reply:
That's tru! haha